android/src/com/unciv/app/MultiplayerTurnCheckWorker.kt

```diff
@@ -0,0 +1,202 @@
+                        PendingIntent.getActivity(applicationContext, 0, notificationIntent, 0)
+                    }
+
+            val notification: NotificationCompat.Builder = NotificationCompat.Builder(applicationContext, NOTIFICATION_CHANNEL_ID_INFO)
+                    .setPriority(NotificationManagerCompat.IMPORTANCE_HIGH) // people are waiting!
+                    .setContentTitle("Unciv - It's your turn!".tr())
+                    .setContentText("Your friends are waiting on your turn.".tr())
+                    .setSmallIcon(R.drawable.uncivicon2)
+                    .setContentIntent(pendingIntent)
+                    .setCategory(NotificationCompat.CATEGORY_SOCIAL)
+                    .setOngoing(false)
+
+            with(NotificationManagerCompat.from(applicationContext)) {
+                notify(NOTIFICATION_ID_INFO, notification.build())
+            }
+        }
+
+        fun startTurnChecker(applicationContext: Context, gameInfo: GameInfo) {
+            val settings = UncivGame.Current.settings
+            if (gameInfo.currentPlayerCiv.playerId == settings.userId) {
+                // May be useful to remind a player that he forgot to complete his turn.
+                notifyUserAboutTurn(applicationContext)
+            } else {
+                showPersistentNotification(applicationContext,
+                        "—", settings.multiplayerTurnCheckerDelayInMinutes.toString())
+                // Initial check always happens after a minute, ignoring delay config. Better user experience this way.
+                enqueue(applicationContext, 0, 1, gameInfo.gameId)
+            }
+        }
+    }
+
+    val delayInMinutes = UncivGame.Current.settings.multiplayerTurnCheckerDelayInMinutes
+
+    override fun doWork(): Result {
+        var failCount = workerParams.inputData.getInt(WORK_DATA_FAIL_COUNT_INT, 0)
+        var gameId = workerParams.inputData.getString(WORK_DATA_GAME_ID)
+        if (null == gameId) {
+            gameId = ""
+        }
+        try {
+            val latestGame = OnlineMultiplayer().tryDownloadGame(gameId)
+            if (latestGame.currentPlayerCiv.playerId == UncivGame.Current.settings.userId) {
+                notifyUserAboutTurn(applicationContext)
+                with(NotificationManagerCompat.from(applicationContext)) {
+                    cancel(NOTIFICATION_ID_SERVICE)
+                }
+            } else {
+                enqueue(applicationContext, 0, delayInMinutes, gameId)
+                updatePersistentNotification()
+            }
+        } catch (ex: Exception) {
+            if (++failCount > 5) {
+                showErrorNotification()
+                with(NotificationManagerCompat.from(applicationContext)) {
+                    cancel(NOTIFICATION_ID_SERVICE)
+                }
+                return Result.failure()
+            } else {
+                enqueue(applicationContext, failCount, delayInMinutes, gameId)
```

Comment:
What I see here: it does a check in 1-15 minutes, however, if the connection is failed, it does a next try in next 1-15 minutes, and does so 5 times.
It means, the user will get notification that something went wrong only in 5-75 minutes.
Maybe, it makes sense to hardcode `delayInMinutes` e.g. as 1 minute until next try in case of failed checks (exceptions) ?

Reply:
Ok, makes sense. Fixed in new commit.